adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/MqttConnectContext.java

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.adapter.mqtt;
+
+import java.util.Objects;
+
+import org.eclipse.hono.util.MapBasedExecutionContext;
+import org.eclipse.hono.util.QoS;
+
+import io.vertx.mqtt.MqttEndpoint;
+
+/**
+ * Contains information required during the processing of an MQTT CONNECT packet.
+ */
+public final class MqttConnectContext extends MapBasedExecutionContext {
+
+    private final MqttEndpoint deviceEndpoint;
+
+    private MqttConnectContext(final MqttEndpoint deviceEndpoint) {
+        this.deviceEndpoint = Objects.requireNonNull(deviceEndpoint);
+    }
+
+    @Override
+    public QoS getRequestedQos() {
+        return QoS.UNKNOWN;
+    }
```

Comment:
maybe we can move this method's definition from *ExecutionContext* to *TelemetryExecutionContext*?